### PR TITLE
Changes in build for new service worker added (#13975)

### LINKED ISSUE #14185: Add start script to locally run extension using chrome extension version MV3
Add script `yarn start:mv3` to locally run application using version 3 of chrome extension.

### PR DESCRIPTION
**Pull request by jpuri**
Fixes #14185

Adding command `yarn start:mv3` to start app locally using MV3

### CODE REVIEW

File: app/manifest/mv3/_base.json [hunk 1]
```diff
@@ -0,0 +1,81 @@
+{
```
Comment (jpuri) [inline app/manifest/mv3/_base.json:1]: PR adds new set of manifest to be used for mv3 at `app/manifest/mv3`.

File: app/manifest/mv3/_base.json [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  "author": "https://metamask.io",
+  "background": {
+    "service_worker": "app-init.js"
+  },
+  "browser_action": {
```
Comment (adonesky1) [inline app/manifest/mv3/_base.json:6]:
```suggestion
  "action": {
```

File: app/scripts/background.js [hunk 3]
```diff
@@ -470,18 +470,21 @@ function setupController(initState, initLangCode) {
...
-    if (count) {
-      label = String(count);
+    if (!process.env.ENABLE_MV3) {
+      let label = '';
+      const count = getUnapprovedTransactionCount();
+      if (count) {
+        label = String(count);
+      }
+      browser.browserAction.setBadgeText({ text: label });
+      browser.browserAction.setBadgeBackgroundColor({ color: '#037DD6' });
```
Comment (adonesky1) [inline app/scripts/background.js:486]:
```suggestion
      browser.action.setBadgeText({ text: label });
      browser.action.setBadgeBackgroundColor({ color: '#037DD6' });
```

Comment (jpuri) [reply app/scripts/background.js:486]: @adonesky1 : will this change work in MV3 also ?

Comment (adonesky1) [reply app/scripts/background.js:486]: Yes these are changes only for MV3: https://developer.chrome.com/docs/extensions/mv3/intro/mv3-migration/#action-api-unification

I wasn't able to get this build to work properly without them...?

File: package.json [hunk 4]
```diff
@@ -10,6 +10,7 @@
     "setup": "yarn install && yarn setup:postinstall",
     "setup:postinstall": "yarn patch-package && yarn allow-scripts",
     "start": "yarn build:dev dev",
+    "start:mv3": "ENABLE_MV3=true yarn build:dev dev",
```
Comment (adonesky1) [inline package.json:13]: here you are setting `ENABLE_MV3=true` should it be `ENABLE_MV3='1'` to match what is expected in `.metamaskrc`. I tried this btw and it seems that setting it to `'1'` makes it work as expected.

File: app/scripts/background.js [hunk 5]
```diff
@@ -470,18 +470,21 @@ function setupController(initState, initLangCode) {
...
-    if (count) {
-      label = String(count);
+    if (process.env.ENABLE_MV3 !== true) {
+      let label = '';
+      const count = getUnapprovedTransactionCount();
+      if (count) {
+        label = String(count);
+      }
+      browser.browserAction.setBadgeText({ text: label });
+      browser.browserAction.setBadgeBackgroundColor({ color: '#037DD6' });
```
Comment (adonesky1) [inline app/scripts/background.js:486]:
```suggestion
      let label = '';
      const count = getUnapprovedTransactionCount();
      if (count) {
        label = String(count);
      }
      if (process.env.ENABLE_MV3 !== true) {
      browser.browserAction.setBadgeText({ text: label });
      browser.browserAction.setBadgeBackgroundColor({ color: '#037DD6' });
      } else. {
       browser.action.setBadgeText({ text: label });
      browser.action.setBadgeBackgroundColor({ color: '#037DD6' });
      }
```

File: package.json (hunk 4 above)
Comment (jpuri) [reply package.json:13]: @adonesky1 : `ENABLE_MV3=true` also works well for me.

**Pull request merged**